packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -130,6 +132,7 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
   AppBar({
```

Comment:
constructor doc should say that automaticallyImplyLeading must not be null (please copy similar text from another constructor to match style), then add an assert that it's not null.